torchaudio/transforms.py

```diff
@@ -1025,3 +1025,127 @@ def forward(self, waveform: Tensor) -> Tensor:
             hp_lifter_freq=self.hp_lifter_freq,
             lp_lifter_freq=self.lp_lifter_freq,
         )
+
+
+class Synth(torch.nn.Module):
```

Comment:
I'd consider splitting this into separate transforms and functionals.

We don't have this kind of setup in torchaudio (monothlic classes with factory methods). Also, not all arguments that are passed to the constructor are necessary for each of the synths (for example square or sin).